pkg/eth/tx_listener.go

```diff
@@ -88,11 +95,23 @@ func (t *TxListener) PollForTxCompletion(txID string, updates chan<- string) {
 
 }
 
-func (t *TxListener) checkTx(hash common.Hash) (bool, error) {
+func (t *TxListener) waitForTimeout() {
+	c := time.After(t.timeout)
+	<-c
+	t.mutex.Lock()
+	t.timeoutPassed = true
+	t.mutex.Unlock()
+}
 
+func (t *TxListener) checkTx(hash common.Hash) (bool, error) {
+	t.mutex.Lock()
```

Comment:
Likely not a real issue in practice so it's up to you here, but the mutex lock is only needed around the timeoutPassed, so we could just wrap the lock/unlock around line 112.  Currently, this will keep locking the mutex even in cases where `TransactionByHash` doesn't return an error.  Given that `TransactionByHash` could also take some time to return from the node, could be locking longer than anticipated.  Again, not sure if this even matters here, but just wanted to flag.